test/rules/no-floating-promises/promises/test.ts.lint

```diff
@@ -0,0 +1,58 @@
+class Promise { }
```

Comment:
add test cases for `returnsPromiseFunction().then()` and calling static Promise functions